### ISSUE TITLE
Authentication using the Service Principal Authentication Method missing in version 3.5.4?

### ISSUE THREAD

**Issue by KorZijl** [microsoft/ARI#224]
Hi,
we are currently using version 3.1.16 with the Service Principal Authentication Method to authenticate to different tenants to retrieve the inventory. In the tenants we've created an Enterprise App with certificate authentication and we've set the appropriate read permissions on the resources. Every weekend we run a scheduled task that calls AzureResourceInventory.ps1 with the **parameters TenantID, AppId and Secret** for all those tenants.

Now we're trying to move from version 3.1.16 to version 3.5.4 but after testing it seems the Service Principal Authentication Method is not available anymore. Executing Invoke-ARI with those parameters returns a lot of "Run Connect-AzAccount to login" messages and the parameters AppId and Secret aren't mentioned anymore in the updated README (https://github.com/microsoft/ARI/tree/main?tab=readme-ov-file#3-parameters).

Did we miss a breaking change alert on this authentication method? And are you willing and planning to reimplement this method?

**Comment by Claudio-Merola** [comment]
Hi @KorZijl 

Our idea was to initially remove the SPN support in the module version, that is why we removed from the README.

But some people have requested that functionality back and it should be working in the version 3.5.4

The parameters should be the same actually, -Appid and -Secret (the secret must be passed as plaintext).

If you are using the parameters correctly and still having issues, please send me the output of the script with the -Debug

**Comment by KorZijl** [comment]
Hi Claudio,
We're using certificate authentication so we pass the PEM-file as the secret value instead of a plaintext secret successfully using version 3.1.16.

I've executed the command below and attached the result as picture (cause the debugging output is messing up the layout):
```powershell
Invoke-ARI -TenantId 419a6976-xxxx-xxxx-xxxx-xxxxxxxxxxxx -AppId 1af2daf0-xxx-xxxx-xxxx-xxxxxxxxxxxx -Secret D:\ARI\AzureResourceInventory-BASE64-ALL.pem -ReportDir D:\ARI\Exports\TESTWORKZ -ReportName ARI_TESTWORKZ -Debug
```
![ARI](https://github.com/user-attachments/assets/3a48493a-bf76-42bb-99eb-ebebec13e516)

I've also tested with Powershell 7.4.5 but the error output is almost similar.

**Comment by KorZijl** [comment]
Hi Claudio,

There are examples to use certificates but they rely on the Windows certificate store (example 7) or on PFX-files (example 9):
https://learn.microsoft.com/en-us/powershell/module/az.accounts/connect-azaccount?view=azps-12.3.0#example-7-connect-using-certificates
https://learn.microsoft.com/en-us/powershell/module/az.accounts/connect-azaccount?view=azps-12.3.0#example-9-connect-using-certificate-file

I need to rethink my authentication method i guess ;)

BTW: For testing purposes i've added a client secret to the enterprise app and used that as plaintext secret but i'm still getting "Run Connect-AzAccount to login" messages.
```powershell
Invoke-ARI -TenantId 419a6976-xxxx-xxxx-xxxx-xxxxxxxxxxxx -AppId 1af2daf0-xxx-xxxx-xxxx-xxxxxxxxxxxx -Secret "x-M8Q~xxxxxxx~xxxx~xx~xxxxxxxxxxxxxxxxxx" -ReportDir D:\ARI\Exports\TESTWORKZ -ReportName ARI_TESTWORKZ -Debug
```
![ARI2](https://github.com/user-attachments/assets/f7011907-0f3c-4852-8328-faeb5b280430)

**Comment by Claudio-Merola** [comment]
Hi @TheOlek 

Are you also using a certificate to authenticate?

**Comment by TheOlek** [comment]
@Claudio-Merola  im passing as plain text

**Comment by Claudio-Merola** [comment]
@TheOlek 

Yes it is, can you run the command adding the -Debug and post the output here?

**Comment by TheOlek** [comment]
@Claudio-Merola 

![Image](https://github.com/user-attachments/assets/6646eecb-4cd3-4f08-8e91-6364004f5a88)

**Comment by TheOlek** [comment]
@Claudio-Merola 

![Image](https://github.com/user-attachments/assets/794d869a-6938-4560-ae04-f836d98091f5)
![Image](https://github.com/user-attachments/assets/57ce71df-b68c-4a0c-96c1-b162d89e7df9)

**Comment by Claudio-Merola** [comment]
@TheOlek 

I think there was a bug there. 

Can you try the version 3.5.6 and see if it works?

**Comment by TheOlek** [comment]
@Claudio-Merola im getting the same behaviour even with the new version

**Comment by Claudio-Merola** [comment]
@TheOlek 

with the version 3.5.5 or 3.5.6?

I released version 3.5.5 and then found that bug, then I just released version 3.5.6 a few minutes after 3.5.5

**Comment by TheOlek** [comment]
@Claudio-Merola  im using 3.5.6

![Image](https://github.com/user-attachments/assets/7f0a91bf-94c4-4e65-9061-7c986dace626)

**Comment by KorZijl** [comment]
I tested this again with 3.5.6 and encountered the same behavior as TheOlek.

**Comment by Claudio-Merola** [comment]
@TheOlek @KorZijl 

I updated the module to the version 3.5.7

Can you guys try it using the -Debug parameter?

This version should at least give more details on what is going on during the authentication phase

**Comment by TheOlek** [comment]
@Claudio-Merola here it is

![Image](https://github.com/user-attachments/assets/322a9d03-4844-499b-ae14-97d7a9598b24)

**Comment by KorZijl** [comment]
@Claudio-Merola: I've got similar output using version 3.5.7:

![Image](https://github.com/user-attachments/assets/264fa2ba-7786-481d-b5af-0388e40aa870)

**Comment by TheOlek** [comment]
@Claudio-Merola  do you have any update on this?

**Comment by Claudio-Merola** [comment]
@TheOlek 

This should be fixed already.  

Can you try to run the following command:

$secret = "YOURSECRET"
$SecurePassword = ConvertTo-SecureString -String $Secret -AsPlainText -Force

If that command runs without issue, then ARI should be running fine, at least without the previous issue.

**Comment by KorZijl** [comment]
Hi @Claudio-Merola 
I've tested this client / secret authentication using version 3.5.14 and can confirm it's fixed now.
If I have an hour to spare I'll try to make a pull request to extend the code with certificate authentication.
